homeassistant/components/roborock/coordinator.py

```diff
@@ -240,7 +284,25 @@ async def _async_update_data(self) -> DeviceProp:
             # Set the new map id from the updated device props
             self._set_current_map()
             # Get the rooms for that map id.
-            await self.set_current_map_rooms()
+
+            # If the vacuum is currently cleaning and it has been IMAGE_CACHE_INTERVAL
+            # since the last map update, you can update the map.
+            if (
+                self.current_map is not None
+                and self.roborock_device_info.props.status.in_cleaning
+                and (
+                    (
+                        dt_util.utcnow() - self.maps[self.current_map].last_updated
+                    ).total_seconds()
```

Comment:
If IMAGE_CACHE_INTERVAL were a `timedelta` then this could save a few lines of code in two places